### PR TITLE
[SPARK-33046][DOCS] Update how to build doc for Scala 2.13 with sbt

### PR DESCRIPTION
**Pull request by sarutak**
### What changes were proposed in this pull request?

This PR fixes the description how to build Spark for Scala 2.13 with sbt.
In the current doc, how to build Spark for Scala 2.13 with sbt is described like:
scala-2 13-build-before

But build fails with this command because scala-2.13 profile is not enabled and scala-parallel-collections is absent.

```
[error] /home/kou/work/oss/spark-scala-2.13/core/src/main/scala/org/apache/spark/rdd/UnionRDD.scala:23: object parallel is not a member of package collection
```

The correct command should be:
```
build/sbt -Pspark-2.13 compile
```

### Why are the changes needed?

The build command is wrong.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

I checked that `sbt -Pspark-2.13` is correct with the following command:
```
build/sbt -Dscala.version=2.13.3 -Phive -Phive-thriftserver -Pyarn -Pkubernetes  compile
```

I also build the modified doc and checked the generated html:
spark-scala-2 13-build-doc-after

### CODE REVIEW

**Review by srowen — APPROVED** [review]

File: docs/building-spark.md [hunk 1]
```diff
@@ -265,15 +265,13 @@ Change the major Scala version using (e.g. 2.13):
 
     ./dev/change-scala-version.sh 2.13
 
-For Maven, please enable the profile (e.g. 2.13):
+Eenable the profile (e.g. 2.13):
```
Comment (dongjoon-hyun) [inline docs/building-spark.md:268]: @sarutak . nit. A typo: `Eenable` -> `Enable`?

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM (except one typo)

File: docs/building-spark.md (hunk 1 above)
Comment (sarutak) [reply docs/building-spark.md:268]: Ah, I'll fix it. Thanks.

**Comment by srowen** [comment]
Merged to master

**Pull request closed without merging**